src/dune_rules/ocaml_flags.ml

```diff
@@ -84,32 +84,36 @@ module Spec = struct
 
   let standard =
     { common = Ordered_set_lang.Unexpanded.standard
-    ; specific = Mode.Dict.make_both Ordered_set_lang.Unexpanded.standard
+    ; specific = Lib_mode.Map.make_all Ordered_set_lang.Unexpanded.standard
     }
 
   let decode =
     let open Dune_lang.Decoder in
     let field_oslu = Ordered_set_lang.Unexpanded.field in
     let+ common = field_oslu "flags"
     and+ byte = field_oslu "ocamlc_flags"
-    and+ native = field_oslu "ocamlopt_flags" in
-    let specific = Mode.Dict.make ~native ~byte in
+    and+ native = field_oslu "ocamlopt_flags"
+    and+ melange = field_oslu "melc_flags" in
```

Comment:
This should be guarded behind the melange syntax. And the name should be `melange.flags` or `melange.melc_flags`

Reply:
Actually, never mind. We have a strange situation here:

1. For the library stanza, we want the `melange.` prefix for the flags
2. For the emit stanza, this prefix is redundant.

Let me think a bit.

Reply:
Okay, so indeed we should have a `melange.` prefix in the library field and in the env field. In the emit stanza, we don't need such a prefix. I'd personally vote for the names `melange.compile_flags` and `compile_flags` in library & emit respectively.